code/modules/mining/machine_vending.dm

```diff
@@ -126,62 +106,89 @@
 /obj/machinery/mineral/equipment_vendor/attack_hand(mob/user)
 	if(..())
 		return
-	interact(user)
+	tgui_interact(user)
 
 /obj/machinery/mineral/equipment_vendor/attack_ghost(mob/user)
-	interact(user)
-
-/obj/machinery/mineral/equipment_vendor/interact(mob/user)
-	user.set_machine(src)
-
-	var/dat
-	dat +="<div class='statusDisplay'>"
-	if(istype(inserted_id))
-		dat += "You have [inserted_id.mining_points] mining points collected. <A href='?src=[UID()];choice=eject'>Eject ID.</A><br>"
+	tgui_interact(user)
+
+/obj/machinery/mineral/equipment_vendor/tgui_data(mob/user)
+	var/list/data[0]
+	. = data
```

Comment:
Best to use `return` when it's possible.
TG has a love for using `. = X` over return. Yet it makes the code a bit more abstract to glance over